widget/contrib/wifi.lua

```diff
@@ -0,0 +1,45 @@
+local wibox     = require("wibox")
+local helpers   = require("lain.helpers")
+
+function factory(args)
+    local wifi      =  { widget = wibox.widget.textbox() }
+    local args      = args or {}
+    local timeout   = args.timeout or 2
+    local settings  = args.settings or function() end
+         
+    function wifi.is_connected()
+        conn_flag = io.popen("[[ ! -z `ping -c 1 -w 3 www.duckduckgo.com` ]] && echo true"):read()
```

Comment:
Never use `io.popen` in Awesome, because it's IO blocking.

Read [here](https://awesomewm.org/doc/api/libraries/awful.spawn.html).